src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -383,8 +380,7 @@ bool StorageKafka::streamToViews()
     else
         in = streams[0];
 
-    std::atomic<bool> stub = {false};
-    copyData(*in, *block_io.out, &stub);
+    copyData(*in, *block_io.out, &stream_cancelled);
```

Comment:
missed that. We can't cancel during copyData, as it's not aware of commits and other kafka-related stuff.
It will be cancelled on underlying layer (kafka buffer)

Reply:
Yep, I already thought about this, but forgot, thanks!

Reply:
But better to add some comment, to avoid further errors